src/types/line.js

```diff
@@ -88,6 +88,7 @@ export default class LineAnnotation extends Element {
 				y = min;
 				y2 = max;
 			}
+			this._horizontal = !scale.isHorizontal();
```

Comment:
Lets not do two things in this function (resolve and set one of the properties, while returning the rest).
You can add the `_horizontal` to the returned object instead. And you might want to move this function out of the class to avoid the //eslint-disable.

Reply:
> Lets not do two things in this function (resolve and set one of the properties, while returning the rest).
> You can add the `_horizontal` to the returned object instead.

Oki even if I thought that all annotations have got own logic to resolve the elements options and the result objetc could be different. But ok!

> You can add the `_horizontal` to the returned object instead. And you might want to move this function out of the class to avoid the //eslint-disable.

It could be static method (or a function of element type). Ok for you?

Reply:
I don't think it can be static, and actually can't be moved out of the class, because its accessed through the class. Lets just leave it as is.

Reply:
I have tried something like that (here LineAnnotation but it's the same for all other types):
```javascript
LineAnnotation.resolveElementProperties = function(chart, options) {
	const scale = chart.scales[options.scaleID];
	let {top: y, left: x, bottom: y2, right: x2} = chart.chartArea;
	let min, max, _horizontal;

	if (scale) {
		min = scaleValue(scale, options.value, NaN);
		max = scaleValue(scale, options.endValue, min);
		if (scale.isHorizontal()) {
			x = min;
			x2 = max;
		} else {
			y = min;
			y2 = max;
		}
		_horizontal = !scale.isHorizontal();
	}
	return {
		x,
		y,
		x2,
		y2,
		width: x2 - x,
		height: y2 - y,
		_horizontal
	};
};
```

and in the annoation.js I invoke it by the element type:

```javascript
const properties = elType.resolveElementProperties(chart, annotation);
```

Make sense? OTherwise I'll revert to previous method of the class.

Reply:
I don't think you can use `elType` to call a static. You would need to use `LineAnnotation`. I disabled the `methods-use-this` rule, so those comments can be removed.
You need to rebase (and `npm i` to get a eslint-plugin-es) to get the new settings.

Reply:
(I might be wrong about the static thing, but I don't really see a reason going that way)

Reply:
> 
> 
> (I might be wrong about the static thing, but I don't really see a reason going that way)

Nop, I'll revert to the method of the class.